iup/eval_projections.py

```diff
@@ -1,5 +1,86 @@
+import polars as pl
 import altair as alt
 
+# get mspe #
+def get_mspe(
+        data_df,
+        pred_df,
+        var
+):
+    """
+    Calculate the mean squared prediction error between the observed
+    data and prediction when both are available. 
+
+    Parameters:
+    --------------
+    data_df: polars.DataFrame
+        The observed vaccine uptake. Must include `date` and the variable-
+        of-interest (var).
+    pred_df: polars.DataFrame
+        The predicted daily uptake. Must include `date` and the variable-
+        of-interest (var).
+    var: string
+        The name of the variable-of-interest.
+    
+    Return:
+    -------------
+    A polars.DataFrame with the MSPE and the date of initializing forecast. 
+    
+    """
+    
+    
+    mspe = data_df.join(
+            pred_df, 
+            on = 'date',
+            how = 'inner'
+        ).with_columns(
+            spe = (pl.col(var) - pl.col(f"{var}_right"))**2
+        ).with_columns(
+            mspe = pl.col('spe').mean(),
```

Comment:
what happens if data are missing in either of the two data frames?

I think you might want to have some checks that the two data frames have the same dates, and only one of each date, eg